src/validations/actionSchema.test.ts

```diff
@@ -0,0 +1,77 @@
+import ValidationError from 'yup/lib/ValidationError';
+
+import { extendLifecycleIdSchema } from './actionSchema';
+
+describe('extend lifecycle ID schema', () => {
+  it('should correctly validate a schema that is valid', () => {
+    expect(() =>
+      extendLifecycleIdSchema.validateSync({
+        newExpirationMonth: '2',
+        newExpirationDay: '15',
+        newExpirationYear: '2023',
+        newScope: 'A new scope',
+        newNextSteps: 'Some new next steps'
+      })
+    );
+  });
+
+  it('should throw a validation error when the month is not between 1 and 12', () => {
+    expect(() =>
+      extendLifecycleIdSchema.validateSync({
+        newExpirationMonth: '-2',
+        newExpirationDay: '15',
+        newExpirationYear: '2023',
+        newScope: 'A new scope',
+        newNextSteps: 'Some new next steps'
+      })
+    ).toThrowError(new ValidationError('Enter a valid expiration date'));
+  });
+
+  it('should throw a validation error when the day is not a possible day in a month', () => {
+    expect(() =>
+      extendLifecycleIdSchema.validateSync({
+        newExpirationMonth: '2',
+        newExpirationDay: '150',
+        newExpirationYear: '2023',
+        newScope: 'A new scope',
+        newNextSteps: 'Some new next steps'
+      })
+    ).toThrowError(new ValidationError('Enter a valid expiration date'));
+  });
+
+  it('should throw a validation error when an invalid year string is entered for the year', () => {
+    expect(() =>
+      extendLifecycleIdSchema.validateSync({
+        newExpirationMonth: '2',
+        newExpirationDay: '15',
+        newExpirationYear: 'abcd',
+        newScope: 'A new scope',
+        newNextSteps: 'Some new next steps'
+      })
+    ).toThrowError(new ValidationError('Enter a valid expiration date'));
+  });
+
+  it('should throw a validation error when a new scope is missing', () => {
+    expect(() =>
+      extendLifecycleIdSchema.validateSync({
+        newExpirationMonth: '2',
+        newExpirationDay: '15',
+        newExpirationYear: '2023',
+        newScope: '',
+        newNextSteps: 'Some new next steps'
+      })
+    ).toThrowError(new ValidationError('Please include a scope'));
+  });
+
+  it('should throw a validation error when next steps are not filled out', () => {
+    expect(() =>
+      extendLifecycleIdSchema.validateSync({
+        newExpirationMonth: '2',
+        newExpirationDay: '15',
+        newExpirationYear: '2023',
+        newScope: 'A new scope',
+        newNextSteps: ''
+      })
+    ).toThrowError(new ValidationError('Please fill out next steps'));
```

Comment:
Rather than testing the specific error message, we _could_ test that the error's `.path` property points to the appropriate field. The test would look something like:
```ts
try {
      extendLifecycleIdSchema.validateSync({
        newExpirationMonth: '2',
        newExpirationDay: '15',
        newExpirationYear: '2023',
        newScope: 'A new scope',
        newNextSteps: ''
      });
      throw new Error('Should not validate successfully');
    } catch (err) {
      expect(err).toBeInstanceOf(ValidationError);
      expect((err as ValidationError).path).toBe('newNextSteps');
    }
```
The good part is that it's not coupled to the specific error message, the bad part is that it's a good bit clunkier because Jest doesn't have a built-in matcher for checking an `Error`'s properties. Up to you if you want to make the change.

Reply:
Nice. I can see how that would save someone a bit of time when changing the error message. It is also a better test because it makes sure the right field is what failed validation. I went ahead and made that change to all of the tests that throw exceptions and pushed it.

Reply:
Done in https://github.com/CMSgov/easi-app/pull/1461/commits/84f99e69085550207bbc29bbe9e002e4407db6b4.